src/datacube.ts

```diff
@@ -213,6 +213,86 @@ export class DataCube {
    * @param {Component} component
    * @returns {Promise<Array<{label: Literal, value: NamedNode}>>}
    */
+  public async componentsValues(components: Component[]):
+    Promise<Map<Component, Array<{label: Literal, value: NamedNode}>>> {
+    const valueBinding = variable("value");
+    const labelBinding = variable("label");
+    const observation = variable("observation");
+    const componentBinding = variable("component");
+
+    const componentIRIs = components.map((comp) => comp.iri);
+
+    const query: SelectQuery = {
+      prefixes,
+      queryType: "SELECT",
+      variables: [componentBinding, valueBinding, labelBinding],
+      distinct: true,
+      from: { default: [namedNode(this.graphIri)], named: [] },
+      where: [
+        {
+          type: "bgp",
+          triples: [
+            {
+              subject: observation,
+              predicate: namedNode("http://www.w3.org/1999/02/22-rdf-syntax-ns#type"),
+              object: cube("Observation"),
+            },
+            {
+              subject: observation,
+              predicate: componentBinding,
+              object: valueBinding,
+            },
+            {
+              subject: observation,
+              predicate: cube("dataSet"),
+              object: namedNode(this.iri),
+            },
+          ],
+        },
+        {
+          type: "filter",
+          expression: {
+            type: "operation",
+            operator: "in",
+            args: [
+              componentBinding,
+              componentIRIs,
+            ],
+          },
+        },
+        ...generateLangOptionals(valueBinding, labelBinding, labelPredicate, this.languages),
+        generateLangCoalesce(labelBinding, this.languages),
+      ],
+      type: "query",
+    };
+
+    const generator = new SparqlGenerator({ allPrefixes: true });
+    const sparql = generator.stringify(query);
+    const rawResult: Array<{component: NamedNode, label: Literal, value: NamedNode}>
+      = await this.fetcher.select(sparql);
+    const resultMap = new Map();
```

Comment:
Would `WeakMap` be more appropriate than `Map` here? Or just an array that stores the values in the same order as the `Component[]` argument? Either way is fine I think …

Reply:
I don't mind using `WeakMap`. Map/WeakMap seem a bit more useful than an array to me.